### PR TITLE
Php3branch > tl1revbranch

### PR DESCRIPTION
**Pull request by php-3**

### CODE REVIEW

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/news/apigetnews.php [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
     curl_setopt_array( $curl, 
                        array(
-                                CURLOPT_URL => "http://ias-systems.com/ifntechmanager/news/apigetiasnews.php",
+//                                CURLOPT_URL => "http://ias-systems.com/ifntechmanager/news/apigetiasnews.php",
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/news/apigetnews.php]: keep this as ias-systems.com even for testing, it is working.

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/products_services/apigetproductsandservices.php [hunk 2]
```diff
@@ -7,21 +7,28 @@
 include_once DOL_DOCUMENT_ROOT . '/custom/extintegration/class/products_services/clsproductandservices.php';
     global $response, $apiPar, $apiHelp;
     
-    $help = array("parameters: " , "apiPar0 > type"  ,"fields: " , "fld0 > rowid " , "fld1 > name" , "fld2 > type" ) ;
+
+    $help = array();
+    $Paramater = array("apiPar0" => type);
+    $Field     = array("fld0" => rowid,"fld1" => name ,"fld2" => type);
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/products_services/apigetproductsandservices.php]: wrong assignment
it should be string

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/products_services/apigetproductsandservices.php [hunk 3]
```diff
@@ -7,21 +7,28 @@
...
+    $help= array("parameters"=>$Paramater, "fields"=>$Field);
 
     $type = $apiPar[0];
+
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/products_services/apigetproductsandservices.php]: if $type is only in use, so why $id and $userid still exist?

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/products_services/apigetproductsandservices.php [hunk 4]
```diff
@@ -7,21 +7,28 @@
...
         $response["error"]  = S_FALSE;
         $response["empty"]  = S_FALSE;
         $response["help"]   = $help;
     }
-    else{
-    if (isset($type)) {
+    else
+    {
+    if (isset($type)) 
+    {
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/products_services/apigetproductsandservices.php]: wrong alignment

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/products_services/apigetproductsandservicesdata.php [hunk 5]
```diff
@@ -7,22 +7,27 @@
...
     global $response, $apiPar, $apiHelp;
     
-    $help = array("parameters: " , "apiPar0 > id"  ,"fields: " , "fld0 > rowid " , "fld1 > name" , "fld2 > datecreation " , "fld3 > type" ,
-        "fld4 > description" , "fld5 > accountancy_code_sell" , "fld6 > accountancy_code_buy" ) ;
-
+    $help = array();
+    $Paramater = array("apiPar0" => id);
+    $Field     = array("fld0" => rowid,"fld1" => name ,"fld2" => datecreation  ,"fld3" => type  ,"fld4" => description  ,"fld5" => accountancy_code_sell  ,"fld6" => accountancy_code_buy);
+    $help= array("parameters"=>$Paramater, "fields"=>$Field);
+
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/products_services/apigetproductsandservicesdata.php]: assignment should be string
please try it and send me JSON output for the help query

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/projects/apigetprojects.php [hunk 6]
```diff
@@ -7,20 +7,24 @@
 include_once DOL_DOCUMENT_ROOT . '/custom/constants.php';
 include_once DOL_DOCUMENT_ROOT . '/custom/extintegration/class/projects/clsprojects.php';
     global $response, $apiPar, $apiHelp;
-    $help = array("parameters:" ,"no parameters" , "fields: " , "fld0 > rowid" ,"fld1 >ref" ,"fld2 >status"  ) ;
+   
+    $help = array();
+    $Paramater = array("no parameters");
+    $Field     = array("fld0" => rowid,"fld1" => ref ,"fld2" => status);
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/projects/apigetprojects.php]: strings

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/projects/apigetprojects.php [hunk 7]
```diff
@@ -7,20 +7,24 @@
...
+    else
+    {
         $objprojects = new clsProjects();
         if (__DEBUG__) {echo '<br>clsProjects object created ';}
         
         $result = $objprojects->fnGetProjects();
-        $response["error"] = "false";
            if(is_null($result))
             {
                 $response["error"] = S_FALSE;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/projects/apigetprojects.php]: incase empty, the JSON must reflect that
$response["empty"] = S_TRUE;

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/proposal/apivalidateproposal.php [hunk 8]
```diff
@@ -1,31 +1,45 @@
...
 include_once DOL_DOCUMENT_ROOT.'/custom/constants.php';
-include_once DOL_DOCUMENT_ROOT.'/custom/extintegration/app/proposal/clsproposal.class.php';
-  $response = array();
-    if($_SERVER['REQUEST_METHOD']=='POST')
+include_once DOL_DOCUMENT_ROOT.'/custom/extintegration/class/proposal/clsproposal.class.php';
+     global $response, $apiPar, $apiHelp;
+    $help = array();
+    $Paramater = array("apiPar0" => id, "apiPar1" => type);
+    $Field     = array("fld0" => rowid,"fld1" => valid_status (1 ,0));
+    $help= array("parameters"=>$Paramater, "fields"=>$Field);
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/proposal/apivalidateproposal.php]: string

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/news/apigetnews.php (hunk 1 above)
Comment (php-3) [reply portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/class/news/apigetnews.php]: i change it to test my changes in localhost

**Pull request merged**